### PR TITLE
Add trace ID to API error message

### PR DESCRIPTION
**Pull request by mattwiller**
The Box API now sends back a second request ID (trace ID) in the
BOX-REQUEST-ID header.  When this value is present, it will be
included in SDK response error messages to aid in troubleshooting.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

File: lib/util/errors.js [hunk 1]
```diff
@@ -10,6 +10,7 @@
 var qs = require('querystring'),
 	httpStatusCodes = require('http-status');
 
+var TRACE_ID_HEADER = 'box-request-id';
```
Comment (nidoran) [inline lib/util/errors.js]:
```suggestion
const TRACE_ID_HEADER_NAME = 'box-request-id';
```

File: lib/util/errors.js [hunk 2]
```diff
@@ -10,6 +10,7 @@
 var qs = require('querystring'),
 	httpStatusCodes = require('http-status');
 
+var TRACE_ID_HEADER_NAME = 'box-request-id';
```
Comment (nidoran) [inline lib/util/errors.js]: const?

**Pull request merged**